workbench/src/menu.c

```diff
@@ -0,0 +1,231 @@
+static void item_new_workbench_activate_cb(GtkMenuItem *menuitem, gpointer user_data)
+{
+	gchar *filename;
+	GError *error = NULL;
+
+	filename = dialogs_create_new_workbench();
+	if (filename == NULL)
+	{
+		return;
+	}
+	wb_globals.opened_wb = workbench_new();
+	workbench_set_filename(wb_globals.opened_wb, filename);
+	if (workbench_save(wb_globals.opened_wb, &error))
+	{
+		menu_set_context(MENU_CONTEXT_WB_CREATED);
+		sidebar_update(SIDEBAR_CONTEXT_WB_CREATED, NULL);
+	}
+	else
+	{
+		dialogs_show_msgbox(GTK_MESSAGE_INFO, _("Could not create new workbench file: %s"), error->message);
+		workbench_free(wb_globals.opened_wb);
+		wb_globals.opened_wb = NULL;
+	}
+	g_free(filename);
+}
+
+
+/* The function handles the menu item "Open workbench" */
+static void item_open_workbench_activate_cb(GtkMenuItem *menuitem, gpointer user_data)
+{
+	gchar *filename;
+	GError *error = NULL;
+
+	filename = dialogs_open_workbench();
+	if (filename == NULL)
+	{
+		return;
+	}
+	wb_globals.opened_wb = workbench_new();
+	if (workbench_load(wb_globals.opened_wb, filename, &error))
+	{
+		menu_set_context(MENU_CONTEXT_WB_OPENED);
+		sidebar_update(SIDEBAR_CONTEXT_WB_OPENED, NULL);
+	}
+	else
+	{
+		dialogs_show_msgbox(GTK_MESSAGE_INFO, _("Could not open workbench file: %s"), error->message);
+		workbench_free(wb_globals.opened_wb);
+		wb_globals.opened_wb = NULL;
+	}
+	g_free(filename);
+}
+
+
+/* The function handles the menu item "Save workbench" */
+static void item_save_workbench_activate_cb(GtkMenuItem *menuitem, gpointer user_data)
+{
+	GError *error = NULL;
+
+	if (wb_globals.opened_wb != NULL)
+	{
+		if (!workbench_save(wb_globals.opened_wb, &error))
+		{
+			dialogs_show_msgbox(GTK_MESSAGE_INFO, _("Could not save workbench file: %s"), error->message);
+		}
+		sidebar_update(SIDEBAR_CONTEXT_WB_SAVED, NULL);
+	}
+}
+
+
+/* The function handles the menu item "Settings" */
+static void item_workbench_settings_activate_cb(GtkMenuItem *menuitem, gpointer user_data)
+{
+	if (wb_globals.opened_wb != NULL)
+	{
+		if (dialogs_workbench_settings(wb_globals.opened_wb))
+		{
+			sidebar_update(SIDEBAR_CONTEXT_WB_SETTINGS_CHANGED, NULL);
+		}
+	}
+}
+
+
+/* The function handles the menu item "Close workbench" */
+static void item_close_workbench_activate_cb(GtkMenuItem *menuitem, gpointer user_data)
+{
+	workbench_free(wb_globals.opened_wb);
+	wb_globals.opened_wb = NULL;
+
+	menu_set_context(MENU_CONTEXT_WB_CLOSED);
+	sidebar_update(SIDEBAR_CONTEXT_WB_CLOSED, NULL);
+}
+
+
+/** Setup the workbench menu.
+ *
+ **/
+gboolean menu_init(void)
+{
+	/* Create menu and root item/label */
+	menu_data.menu = gtk_menu_new();
+	menu_data.root_item = gtk_menu_item_new_with_label(_("Workbench"));
+	gtk_widget_show(menu_data.root_item);
+
+	/* Create new menu item "New Workbench" */
+	menu_data.item_new = gtk_menu_item_new_with_mnemonic(_("_New"));
+	gtk_widget_show(menu_data.item_new);
+	gtk_menu_append(GTK_MENU (menu_data.menu), menu_data.item_new);
```

Comment:
[`gtk_menu_append()`](https://developer.gnome.org/gtk2/stable/GtkMenu.html#gtk-menu-append) is deprecated for a long time in favor to [`gtk_menu_shell_append()`](https://developer.gnome.org/gtk2/stable/GtkMenuShell.html#gtk-menu-shell-append), and its use [breaks compilation under GTK3](https://travis-ci.org/geany/geany-plugins/jobs/266322308#L4962) which removed the former.